integration-libs/cds/src/components/recent-searches/recent-searches.service.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { interval, last, of, ReplaySubject } from 'rxjs';
+import { concatMap, map, take, takeWhile } from 'rxjs/operators';
+import { WindowRef } from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class RecentSearchesService {
+  private readonly recentSearchesSource = new ReplaySubject<string[]>();
+  public recentSearches$ = this.recentSearchesSource.asObservable();
+  constructor(protected winRef: WindowRef) {
+    this.addRecentSearchesListener();
+  }
+
+  checkAvailability() {
+    return interval(250).pipe(
```

Comment:
In SSR this interval here will slow down the rendering of the page by 2.5s -> 250ms * 100 (take(100)). This delay is likely to cause the SSR engine to timeout and fail to render the page altogether. I would suggest to wrap the whole function `addRecentSearchesListener` in `this.winRef.isBrowser()`.

```typescript
if (this.winRef.isBrowser()) {
this.addRecentSearchesListener();
}
```